syntax/compile_scanner_test.go

```diff
@@ -0,0 +1,85 @@
+package syntax
+
+import (
+	"testing"
+
+	"github.com/arr-ai/wbnf/parser"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func assertCompileScanner(t *testing.T, source string) bool { //nolint:unparam
+	pc := ParseContext{SourceDir: ".."}
+	ast, err := pc.Parse(parser.NewScanner(" " + source + " "))
+	require.NoError(t, err, "%s", source)
+	expr := pc.CompileExpr(ast)
+	return assert.Equal(t, source, expr.Source().String())
+}
```

Comment:
I'm not very sure what the test does, is it supposed to see whether a literal's scanner is the whole literal?

Reply:
Added explanatory comments.